tck/src/main/java/org/eclipse/microprofile/config/tck/ArrayConverterTest.java

```diff
@@ -1654,120 +1658,161 @@ public void testInstantSetInjection() {
+        Assert.assertTrue(IntStream.range(0, expectedValue.length).allMatch(i -> {
+        try {
+                return expectedValue[i].toURI().equals(value[i].toURI());
+        } 
+        catch (URISyntaxException e) {
+                e.printStackTrace();
+        }
+        return false;
+        }));
+    }
+
+    private void assertURLListEquals(List<URL> value) throws MalformedURLException {
+        List<URL> expectedValue = Arrays.asList(new URL[]{
+            new URL("http://microprofile.io"),
+            new URL("http://openliberty.io"),
+            new URL("http://microprofile.io")});
+        Assert.assertTrue(IntStream.range(0, expectedValue.size()).allMatch(i -> {
+        try {
+                return expectedValue.get(i).toURI().equals(value.get(i).toURI());
+        } 
+        catch (URISyntaxException e) {
+                e.printStackTrace();
+        }
+        return false;
+        }));
+    }
+
+
     @Test
-    public void testUrlArrayLookupProgrammatically() throws MalformedURLException {
+    public void testUrlArrayLookupProgrammatically() throws MalformedURLException, URISyntaxException {
         URL[] value = config.getValue("tck.config.test.javaconfig.converter.urlvalues", URL[].class);
         Assert.assertNotNull(value);
         Assert.assertEquals(value.length, 3);
-        Assert.assertEquals(value, new URL[]{
-            new URL("http://microprofile.io"),
-            new URL("http://openliberty.io"),
-            new URL("http://microprofile.io")});
+        assertURLArrayEquals(value);
         URL[] single = config.getValue("tck.config.test.javaconfig.converter.urlvalue", URL[].class);
         Assert.assertNotNull(single);
         Assert.assertEquals(single.length, 1);
-        Assert.assertEquals(single, new URL[]{ new URL("http://microprofile.io") });
+        Assert.assertEquals(single[0].toURI(), new URL[]{ new URL("http://microprofile.io") }[0].toURI());
     }
 
     @Test
-    public void testGetUrlArrayConverter() throws MalformedURLException {
+    public void testGetUrlArrayConverter() throws MalformedURLException, URISyntaxException {
         URL[] value = config.getConverter(URL[].class).get().convert("http://microprofile.io,http://openliberty.io,http://microprofile.io");
         Assert.assertNotNull(value);
         Assert.assertEquals(value.length, 3);
-        Assert.assertEquals(value, new URL[]{
-            new URL("http://microprofile.io"),
-            new URL("http://openliberty.io"),
-            new URL("http://microprofile.io")});
+        assertURLArrayEquals(value);
         URL[] single = config.getConverter(URL[].class).get().convert("http://microprofile.io");
         Assert.assertNotNull(single);
         Assert.assertEquals(single.length, 1);
-        Assert.assertEquals(single, new URL[]{ new URL("http://microprofile.io") });
+        Assert.assertEquals(single[0].toURI(), new URL[]{ new URL("http://microprofile.io") }[0].toURI());
```

Comment:
Similarly to the extracted method to assert an array with 3 elements it seems worth it to extract one for the array with single element (including the null and length check) as it is a reoccurring assertion pattern.

Reply:
ok. I will refactor.